datalad/core/distributed/tests/test_clone.py

```diff
@@ -499,3 +506,64 @@ def test_relative_submodule_url(path):
     eq_(subinfo['gitmodule_url'],
         # must be a relative URL, not platform-specific relpath!
         '{}/{}'.format(op.pardir, 'origin'))
+
+
+def test_decode_source_spec():
+    # resolves datalad RIs:
+    eq_(decode_source_spec('///subds'),
+        dict(source='///subds', giturl=consts.DATASETS_TOPURL + 'subds', version=None,
+             type='dataladri', default_destpath='subds'))
+    assert_raises(NotImplementedError, decode_source_spec,
+                  '//custom/subds')
+
+    # doesn't harm others:
+    for url in (
+            'http://example.com',
+            '/absolute/path',
+            'file://localhost/some',
+            'localhost/another/path',
+            'user@someho.st/mydir',
+            'ssh://somewhe.re/else',
+            'git://github.com/datalad/testrepo--basic--r1',
+    ):
+        props = decode_source_spec(url)
+        dest = props.pop('default_destpath')
+        eq_(props, dict(source=url, version=None, giturl=url, type='giturl'))
+
+    # RIA URIs with and without version specification
+    dsid = '6d69ca68-7e85-11e6-904c-002590f97d84'
+    for proto, loc, version in (
+            ('http', 'example.com', None),
+            ('http', 'example.com', 'v1.0'),
+            ('http', 'example.com', 'some_with@in_it'),
+            ('ssh', 'example.com', 'some_with@in_it'),
+    ):
+        spec = 'ria+{}://{}{}{}'.format(
+            proto,
+            loc,
+            '#{}'.format(dsid),
+            '@{}'.format(version) if version else '')
+        eq_(decode_source_spec(spec),
+            dict(
+                source=spec,
+                giturl='{}://{}{}{}/{}'.format(
+                    proto,
+                    loc,
+                    ':' if proto == 'ssh' else '/',
+                    dsid[:3],
+                    dsid[3:]),
+                version=version,
+                default_destpath=dsid,
+                type='ria')
+        )
+    # not a dataset UUID
+    assert_raises(ValueError, decode_source_spec, 'ria+http://example.com#123')
+
+
+@skip_if_no_network
+@with_tempfile()
+def test_ria_http(path):
+    # can we clone from the store w/o any dedicated config
+    ds = clone('ria+http://store.datalad.org#{}'.format(datalad_store_testds_id), path)
+    ok_(ds.is_installed())
+    eq_(ds.id, datalad_store_testds_id)
```

Comment:
- special remote needs to be autoenabled
- origin needs annex-ignore
- ideally we should probably set publish-depends for origin
- test availability via whereis?

Reply:
Re availability: May be test, that fsck doesn't spit out some (fixed loaction log ...)?

Reply:
- `get` needs work. 
- May be also include some annexed content in archive?

Reply:
Thx!

I will not test `get`. A get would require `ria-remote`. And we can test a get in the ria-remote tests for the main store. The test dataset already contains an annexed file.

If we decide to suck ria-remote into datalad-core, adding a test would be trivial, and can be done

Reply:
> The test dataset already contains an annexed file.

Yes, but we should prob. put an archive in there as well. However, we can do that within git-annex-ria-remote tests for now - agreed.

Reply:
That would require SSH access to the main store. Not a good idea, IMHO

Reply:
I have added a new test that goes through all the steps for a dataset with a subdataset and ensures a proper setup for both